package.json

```diff
@@ -2,12 +2,16 @@
   "name": "@dintero/email-chk",
   "version": "0.0.0-development",
   "description": "",
-  "main": "dist/index.js",
-  "types": "dist/index.d.ts",
+  "main": "lib/cjs/index.js",
+  "module": "lib/esm/index.js",
```

Comment:
Why not use export?

```diff
diff --git a/package.json b/package.json
index cd01a9f..4072883 100644
--- a/package.json
+++ b/package.json
@@ -2,9 +2,13 @@
   "name": "@dintero/email-chk",
   "version": "0.0.0-development",
   "description": "",
-  "main": "lib/cjs/index.js",
-  "module": "lib/esm/index.js",
   "types": "lib/types/index.d.ts",
+  "type":"module",
+  "main": "./lib/cjs/index.js",
+  "exports": {
+    "import": "./lib/esm/index.js",
+    "require": "./lib/cjs/index.js"
+  },
   "author": "Thoralf Thelle <thoralf@dintero.com>",
   "license": "MIT",
   "private": false,
@@ -44,8 +48,8 @@
     "clean": "rm -rf ./lib",
     "check": "yarn run lint && yarn run test",
     "build": "yarn run clean && yarn run build:esm && yarn run build:cjs",
-    "build:esm": "tsc -p ./tsconfig.esm.json && echo '{ \"type\": \"module\" }' > ./lib/esm/package.json",
-    "build:cjs": "tsc -p ./tsconfig.cjs.json && echo '{ \"type\": \"commonjs\" }' > ./lib/cjs/package.json",
+    "build:esm": "tsc -p ./tsconfig.esm.json",
+    "build:cjs": "tsc -p ./tsconfig.cjs.json",
     "lint": "eslint ./src",
     "release": "semantic-release",
     "prepublishOnly": "yarn run build"
```

and do we need to generate a package.json for containing just type?

Reply:
`tsc` doesn't emit`.mjs` for ESM (unless the source is `.mts`), which leads to need for a `{ type: module }` package.json to state that this is an ESM module. `{ type: commonjs }` is done for the CJS module since the parent uses a different type. I don't think we need to do it this way, but not having this broke ESM compatibility during testing.